tests/fixtures/rules/infinite-reactive-loop/valid/test08-input.svelte

```diff
@@ -0,0 +1,16 @@
+<script>
+  let a = 0
+
+  const func = () => {
+    return {
+      then: (fun) => fun(),
+      catch: (fun) => fun(),
+    }
+  }
+
+  $: (() => {
+    // TODO: This is a false positive. we need to check if the `then` or `catch` is `Promise` object's function.
```

Comment:
@ota-meshi 

I need your help!
I wanted to check that `then` or `catch` came from `Promise` type, but I couldn't find a way.
Could you please tell me?

Reply:
I think it's difficult to strictly check whether it's a Promise object or not. However, it's mostly okay to consider objects with `then` and `catch` methods that take functions as parameters to be Promises.
Other popular plugins also judge that way.

- https://github.com/typescript-eslint/typescript-eslint/blob/4ab9bd7f7c1812df4371d1fd3202969c1039e8a7/packages/eslint-plugin/src/rules/no-floating-promises.ts#L234
- https://github.com/eslint-community/eslint-plugin-promise/blob/6d046e6033d43d2c023966b33a7fabc1438899c2/rules/lib/is-promise.js#L14

It's clearly a false positive in this test case, but I think we can consider that people don't do it.